backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteNonRootBranch.java

```diff
@@ -56,10 +56,10 @@ public GitMacheteNonRootBranch(
 
     // Note: since the class is final, `this` is already @Initialized at this point.
 
-    // This is a hack necessary to create an immutable cyclic structure (children pointing at the parent & parent
-    // pointing at the children).
+    // This is a hack necessary to create an immutable cyclic structure
```

Comment:
I can see the same fragment of code for `GitMacheteRootBranch`... could we move it to base class? :thinking: